File: perl/2_files.pl
#!/usr/bin/perl -w
use strict;

#`<` means "read", `>` means "write", `>>` means "append"
#Just like in UNIX, you can read in a file to another program with `<`
#or you can redirect the output of one program to a file like this: `ps > out.txt`
#where `ps` is a program, and "out.txt" is the file created and written to.

#Open filehandle for reading, where FH is an alias for the filehandle that is created
#This is a relative path to the file. Google "relative path vs. absolute path"
#if you don't know what they are. A relative path is based on the current/present working
#directory (type `pwd` on the command line to find the Present Working Directory or PWD)
open FH, "<", "2_files/temp.txt";
while (<FH>){
    print $_;#In this context, $_ turns into an alias for a line in the file
    print "\n";
}
close FH;#You *need* to close the filehandle when you're done with it,
         #or if you want to change to another access method (i.e. reading vs writing).

#Usually the above while loop syntax is used, but this works, too:
open FH, "<", "2_files/temp.txt";
foreach my $line (<FH>){
    print $line;
    print "\n";
}
close FH;

open FH, ">", "2_files/temp.txt";
print FH "Overwrite the file with this string.\n";
close FH;

open FH, ">>", "2_files/temp.txt";
print FH "Append this string to the file.\n";
close FH;

File: python/2_files.py
#!/usr/bin/env python
a_string = 'This is a string. Wow!\n'
another_string = 'Whoa, this is another string!\n'

file_location = '2_files/temp.txt'

#You must open and close data streams (aka "file handles") to files:
#THINK: What is `output`?
#ANSWER: It is a "file object". It's not the file itself. It represents the file.
output = open( file_location , 'w' ) #`w` means "write mode"

output.write( a_string )
output.write( another_string )
#The filehandle will not close unless the parentheses are present on `close`:
output.close() 

print "This is what's in the file so far:"

input = open( file_location, 'r' ) #`r` means "read mode"
for file_line in input:
    print file_line
input.close()

